website/src/views/timetable/ShareTimetable.tsx

```diff
@@ -34,8 +38,13 @@ type State = {
   shortUrl: string | null;
 };
 
-function shareUrl(semester: Semester, timetable: SemTimetableConfig): string {
-  return absolutePath(timetableShare(semester, timetable));
+function shareUrl(
+  semester: Semester,
+  timetable: SemTimetableConfig,
+  colors?: ColorMapping | null,
+  themeId?: ThemeId | null,
```

Comment:
Optional parameters should not be `| null` 

```suggestion
  colors?: ColorMapping,
  themeId?: ThemeId,
```